main.py

```diff
@@ -1,29 +1,83 @@
 """
 This is an Add-On for DocumentCloud which wraps the tabula-py library
 https://github.com/chezou/tabula-py
-
-It attempts to convert tables from PDFs to CSVs
+It allows you to export tables from a set of PDFs into CSVs and
+download the resulting CSVs as a zip file.
 """
 
+import os
+import sys
+import zipfile
+import subprocess
+from urllib.parse import urlparse
+
+import requests
 import tabula
+
 from documentcloud.addon import AddOn
+from clouddl import *
+from clouddl.clouddl import dropbox_url, gdrive_url
 
 
 class Tabula(AddOn):
     """A tabula Add-On for DocumentCloud"""
 
-    def main(self):
+    def fetch_files(self, url):
+        """Fetch the files from either a cloud share link or any public URL"""
+        self.set_message("Downloading the files...")
 
-        document = self.client.documents.get(self.documents[0])
-        with open(f"{document.slug}.pdf", "wb") as pdf_file:
-            pdf_file.write(document.pdf)
+        os.makedirs(os.path.dirname("./out/"), exist_ok=True)
+        cloud_urls = [dropbox_url, gdrive_url]
+        if any(cloud_url in url for cloud_url in cloud_urls):
+            # surpress output during download to avoid leaking
+            # private information
+            stdout = sys.stdout
+            sys.stdout = open(os.devnull, "w")
+            grab(url, "./out/")
+            # restore stdout
+            sys.stdout = stdout
+            rename_file = "cd out; mv * template.json"
```

Comment:
This seems problematic - What if `grab` grabs more than one file?  Also I believe there are python wrappers to renaming files which would be preferably to calling out the external commands.